zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/routing/DbRoutingState.java

```diff
@@ -65,4 +63,15 @@ public void initializeRoutingInfo(final int partitionCount) {
     currentRoutingInfo.setMessageCorrelation(new MessageCorrelation.HashMod(partitionCount));
     columnFamily.insert(key, currentRoutingInfo);
   }
+
+  @Override
+  public void setDesiredPartitions(final Set<Integer> partitions) {
+    final var currentMessageCorrelation = messageCorrelation();
```

Comment:
Perhaps a better name would be message correlation _strategy_. That name is actually used internally. It describes how the engine correlates messages.

Reply:
I do prefer that over the exact same name 😄

Reply:
To be clear: This _is_ the same concept. The engine _does_ message correlation and the routing state _describes_ how message correlation is done.

Reply:
Thanks for clearing that up. That makes a lot of sense now 👍 I definitely prefer `MessageCorrelationStrategy`. It's more descriptive imo and makes sure it can't be confused with a message correlation itself.